plugin/agent/workloadattestor-k8s/k8s.go

```diff
@@ -0,0 +1,188 @@
+	} `json:"items"`
+}
+
+const (
+	selectorType string = "k8s"
+)
+
+func (p *k8sPlugin) Attest(req *workloadattestor.AttestRequest) (*workloadattestor.AttestResponse, error) {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	log.Printf("Attesting PID: %v", req.Pid)
+	resp := workloadattestor.AttestResponse{}
+
+	httpResp, err := p.httpClient.Get(fmt.Sprintf("http://localhost:%v/pods", p.kubeletReadOnlyPort))
+	if err != nil {
+		return &resp, err
+	}
+	defer httpResp.Body.Close()
+	respBytes, err := ioutil.ReadAll(httpResp.Body)
+
+	var podInfo *pod
+	err = json.Unmarshal(respBytes, &podInfo)
+	if err != nil {
+		return &resp, err
+	}
+
+	cgroups, err := getCgroups(fmt.Sprintf("/proc/%v/cgroup", req.Pid), p.fs)
+	if err != nil {
+		return &resp, err
+	}
+
+	var containerID string
+	for _, cgroup := range cgroups {
+		// We are only interested in kube pods entries
+		if len(cgroup[2]) < 9 {
+			continue
+		}
+
+		substring := cgroup[2][:9]
+		if substring == "/kubepods" {
+			parts := strings.Split(cgroup[2], "/")
+
+			if len(parts) < 5 {
+				log.Printf("Kube pod entry found, but without container id: %v", substring)
+				continue
+			}
+			containerID = parts[4]
+			break
+		}
+	}
+
+	for _, item := range podInfo.Items {
+		for _, status := range item.Status.ContainerStatuses {
+			containerURL, err := url.Parse(status.ContainerID)
+			if err != nil {
+				return &resp, err
+			}
+
+			if containerID == containerURL.Host {
+				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("serviceAccountName:%v", item.Spec.ServiceAccountName)})
+				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("namespace:%v", item.Metadata.Namespace)})
+				log.Printf("Selectors found: %v", resp.Selectors)
+				return &resp, err
+			}
+		}
+	}
+
+	log.Print("No selectors found")
+	return &resp, err
+}
+
+func getCgroups(path string, fs fileSystem) (cgroups [][]string, err error) {
+	// http://man7.org/linux/man-pages/man7/cgroups.7.html
+	// https://www.kernel.org/doc/Documentation/cgroup-v2.txt
+
+	file, err := fs.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		token := scanner.Text()
+		substrings := strings.SplitN(token, ":", 3)
+		if len(substrings) < 3 {
+			return cgroups, fmt.Errorf("cgroup entry contains %v colons, but expected at least two colons: %v", len(substrings), token)
+		}
+		cgroups = append(cgroups, substrings)
+	}
+
+	return cgroups, err
+}
+
+func (p *k8sPlugin) Configure(req *spi.ConfigureRequest) (*spi.ConfigureResponse, error) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	resp := &spi.ConfigureResponse{}
+
+	// Parse HCL config payload into config struct
+	config := &k8sPluginConfig{}
+	hclTree, err := hcl.Parse(req.Configuration)
+	if err != nil {
+		resp.ErrorList = []string{err.Error()}
+		return resp, err
+	}
+	err = hcl.DecodeObject(&config, hclTree)
```

Comment:
I checked in a shared lib to do this since I found myself doing it so often. Feel free to use it if you like, here: https://github.com/spiffe/spire/blob/master/pkg/common/config/config.go

Reply:
Oops I guess that one takes a file... it could be easily updated though without breaking existing call sites if you were so inclined